### PR TITLE
Apply field length normalisation to title in PostgreSQL search 

### PR DESCRIPTION
**Pull request by kaedroho**
This improves the ranking of results from PostgreSQL by increasing the score of matches on terms in the title if the title of the page is short. This has the effect of making more specific results rank higher than less specific ones.

### CODE REVIEW

**Comment by lb-** [comment]
@kaedroho  - just to keep on top of what needs triage, we are marking all WIP PRs as a draft

File: wagtail/contrib/postgres_search/backend.py [hunk 1]
```diff
@@ -111,6 +125,7 @@ def autocomplete(self):
         texts = []
         for field in self.search_fields:
             for current_field, boost, value in self.prepare_field(self.obj, field):
+                # FIXME: I know it's wrong to exclude title here. We're not demoing htis and the autocomplete PR should set this right
```
Comment (kaedroho) [inline wagtail/contrib/postgres_search/backend.py:131]: I'll sort this once autocomplete is on master

File: wagtail/search/tests/test_queries.py [hunk 2]
```diff
@@ -221,3 +222,115 @@ def test_two_filters_and_query(self):
...
+        self.assertEqual(balanced_reduce(add, SequenceClass(1)), 0)
+        self.assertEqual(balanced_reduce(add, SequenceClass(1), 42), 42)
+
+        d = {"one": 1, "two": 2, "three": 3}
+        self.assertEqual(balanced_reduce(add, d), "".join(d.keys()))
+
+    # This test is specific to balanced_reduce
+    def test_is_balanced(self):
+        # Tests that balanced_reduce returns the object as a balanced tree
+        class Node:
```
Comment (kaedroho) [inline wagtail/search/tests/test_queries.py:317]: Just realised this class is completely unnecessary. I could just use the letter strings as the nodes... :facepalm:

File: wagtail/contrib/postgres_search/backend.py [hunk 3]
```diff
@@ -96,6 +96,19 @@ def id(self):
         """
         return force_str(self.obj.pk)
 
+    @cached_property
+    def title(self):
+        """
+        Returns all values to index as "title". This is the value of all SearchFields that have title=True
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail/contrib/postgres_search/backend.py:106]: Sorry, that was an early version, I forgot to update the docstring!

**Comment by kaedroho** [comment]
docstrings updated!

**Pull request merged**